src/Storage/Node.cs

```diff
@@ -2962,6 +3145,37 @@ private void DoCopy(string targetPath, string newName)
                 }
             }
         }
+
+        private Node DoCopyAndGetCopy(string targetPath, string newName)
+        {
+            bool first = true;
+            var sourcePath = this.Path;
+            Node copyOfSource = null;
+            if (!Node.Exists(sourcePath))
+                throw new ContentNotFoundException(sourcePath);
+            foreach (var sourceNode in NodeEnumerator.GetNodes(sourcePath, ExecutionHint.ForceRelationalEngine))
+            {
+                var targetNodePath = targetPath + sourceNode.Path.Substring(sourcePath.Length);
+                targetNodePath = RepositoryPath.GetParentPath(targetNodePath);
+                var targetNode = Node.LoadNode(targetNodePath);
+                if (first)
+                {
+                    copyOfSource = sourceNode.MakeCopy(targetNode, newName);

```

Comment:
Please consider simplifying this, change the original methods as little as possible. E.g. filling the target in an if block, instead of duplicating the whole block.